src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private static final int EMPTY = 0;
+    private static final double GROW = 1.5;
+    private int size;
+    private int capacity;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_CAPACITY];
+        size = EMPTY;
+        capacity = START_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("IndexOutOfBounds");
+        } else {
+            if (size == capacity) {
+                grow();
+            }
+            T[] newArray = (T[]) new Object[capacity];
+            System.arraycopy(array, 0, newArray,0, index);
+            newArray[index] = value;
+            System.arraycopy(array, index, newArray,index + 1, size - index);
+            array = newArray;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() > capacity) {
+            grow();
+        }
```

Comment:
you have your list with capacity 10, user wants to add all values from list of 100 elements
will 1 grow be enough for this operation?

Please make more universal solution, most easy one is to reuse add(T value) in cycle and let in rescale array when required